src/mpid/ch4/src/ch4r_callbacks.h

```diff
@@ -456,23 +456,22 @@ static inline int MPIDI_send_target_msg_cb(int handler_id, void *am_hdr,
         rreq = MPIDI_CH4U_dequeue_posted(hdr->src_rank, hdr->tag, hdr->context_id,
                                          &MPIDI_CH4U_COMM(root_comm, posted_list));
 #else /* MPIDI_CH4_DIRECT_NETMOD */
-        while (TRUE) {
+        int continue_matching = 1;
+        while (continue_matching) {
             rreq = MPIDI_CH4U_dequeue_posted(hdr->src_rank, hdr->tag, hdr->context_id,
```

Comment:
Can this change also be applied to other wildcard_recv handling code in the callback file ? Actually, why not just squash them with the previous commit ?